compilation: msgfmt uses unexistant file (probably wrong flags)

### Affected Version

trying to install `yay 12.4.2-1`
having `msgfmt (GNU gettext-tools) 0.22.5`

<!-- Please ensure you are using the latest yay-git package
Use `yay -V` to get installed version
Example: `yay v8.1139.r0.g9ac4ab6 - libalpm v11.0.1` -->

### Describe the bug

```
msgfmt po/ca.po -o po/ca.mo
msgfmt: error while opening "-o" for reading: No such file or directory
make: *** [Makefile:134: po/ca.mo] Error 1
```

### Reproduction Steps

1. Do as it is said in README.md
```bash
sudo pacman -S --needed git base-devel && \
    git clone https://aur.archlinux.org/yay.git && \
    cd yay && \
    makepkg -si
```

### Expected behavior

yay installs

### Output

<!--
Include the FULL output of any relevant commands/configs
The current yay config can be printed with `yay -Pg`
Paste services are only needed for excessive output (>500 lines)
Use --debug to add pacman and yay debug logs 
or add the following key to your ~/.config/yay/config.json to only get yay debug logs
{
  "debug": true
}
-->

```sh
$ makepkg -si
==> Making package: yay 12.4.2-1 (Sat 05 Oct 2024 09:51:35 PM MSK)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found yay-12.4.2.tar.gz
==> Validating source files with sha256sums...
    yay-12.4.2.tar.gz ... Passed
==> Extracting sources...
  -> Extracting yay-12.4.2.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
make: Nothing to be done for 'build'.
==> Entering fakeroot environment...
==> Starting package()...
msgfmt po/ca.po -o po/ca.mo
msgfmt: error while opening "-o" for reading: No such file or directory
make: *** [Makefile:134: po/ca.mo] Error 1
==> ERROR: A failure occurred in package().
    Aborting...
```

--- a/Makefile
+++ b/Makefile
@@ -131,4 +131,4 @@ locale:
 	done
 
 ${LOCALEDIR}/%.mo: ${LOCALEDIR}/%.po
-	msgfmt $< -o $@
+	msgfmt -o $@ $<

